File: python/vista_api_client/api_resources/roles.py

from vista_api_client.api_resources.api_resource import ApiResource, HttpMethods


class Roles(ApiResource):
    def list(self, org_id='*'):
        return self.dispatch('/v1/roles', HttpMethods.GET, {
            'org_id': org_id,
            'branch': self.branch,
        })

    def upsert(self, role_id, permissions, owners={}, parent_roles=None, org_id='*'):
        parents = parent_roles or []

        return self.dispatch('/v1/roles', HttpMethods.POST, {
            'id': role_id,
            'permissions': permissions,
            'owners': owners,
            'parent_roles': parents,
            'org_id': org_id,
            'branch': self.branch,
        })

File: js/vista-api-client/src/resources/roles.js

import { HTTP_METHODS, ApiResource } from '../apiResource.js';

class Roles extends ApiResource {
    list = async (orgId='*') => {
        return this.dispatch('/v1/roles', HTTP_METHODS.GET, {
            org_id: orgId,
            branch: this.branch,
        });
    }

    upsert = async (roleId, resourceTypeToAttributeToActions, parentRoles=[], orgId='*') => {
        return this.dispatch('/v1/roles', HTTP_METHODS.POST, {
            id: roleId,
            resource_type_to_attribute_to_actions: resourceTypeToAttributeToActions,
            parent_roles: parentRoles,
            org_id: orgId,
            branch: this.branch,
        });
    }
}

export default Roles;
